Mathlib/NumberTheory/SmoothNumbers.lean

```diff
@@ -287,6 +287,16 @@ lemma smoothNumbers_succ {N : ℕ} (hN : ¬ N.Prime) : N.succ.smoothNumbers = N.
 @[gcongr] lemma smoothNumbers_mono {N M : ℕ} (hNM : N ≤ M) : N.smoothNumbers ⊆ M.smoothNumbers :=
   fun _ hx ↦ ⟨hx.1, fun p hp => (hx.2 p hp).trans_le hNM⟩
 
+/-- All $m$, $0 < m < n$ are $n$-smooth numbers -/
+lemma mem_smoothNumbers_of_lt (m n : ℕ) (hm : 0 < m) (hmn : m < n) : m ∈ n.smoothNumbers := by
+  rw [smoothNumbers_eq_factoredNumbers]
+  constructor
+  · exact not_eq_zero_of_lt hm
+  · intro p; intro h
+    apply Nat.le_of_mem_factors at h
+    rw [Finset.mem_range]
+    exact Nat.lt_of_le_of_lt h hmn
```

Comment:
```suggestion
/-- All `m`, `0 < m < n` are `n`-smooth numbers -/
lemma mem_smoothNumbers_of_lt {m n : ℕ} (hm : 0 < m) (hmn : m < n) : m ∈ n.smoothNumbers :=
  smoothNumbers_eq_factoredNumbers _ ▸ ⟨not_eq_zero_of_lt hm,
  fun _ h => Finset.mem_range.mpr <| lt_of_le_of_lt (le_of_mem_factors h) hmn⟩
```
I suggest the following changes:
- Using backticks instead of `$`.
- Using implicit parameters for `n` and `m`.
- Term mode proof.